pkg/addons/applier.go

```diff
@@ -341,6 +365,29 @@ func mcCredentialsEnvVars(s *state.State) ([]byte, error) {
 	return credsEnvVarsMC, nil
 }
 
+func credentialsHash(s *state.State, credsType credentials.Type) (string, error) {
+	creds, err := credentials.ProviderCredentials(s.Cluster.CloudProvider, s.CredentialsFilePath, credsType)
+	if err != nil {
+		return "", err
+	}
+
+	omap := orderedmap.New()
+	for k, v := range creds {
+		omap.Set(k, v)
+	}
+	omap.SortKeys(sort.Strings)
+
+	hash := fmt.Sprintf("kubeone-%s", s.Cluster.CloudProvider.CloudProviderName())
+	for k, v := range creds {
```

Comment:
should be iteration over the omap, not creds

Reply:
Fixed.